src/nvim/message.c

```diff
@@ -323,6 +323,14 @@ MsgID msg_multihl(MsgID id, HlMessage hl_msg, const char *kind, bool history, bo
     }
   }
 
+  // progress message are special displayed as "title: msg...percent%"
+  if (is_progress && msg_data && msg_data->title.size != 0) {
```

Comment:
How does a `ext_messages` UI override this behavior?

Reply:
I am asking the opposite: I don't see where this is gated. It sounds like it is though.